src/users/context/users.tsx

```diff
@@ -0,0 +1,255 @@
+
+export type Props = {
+  children: JSX.Element
+}
+
+export interface UsersContextType {
+  draftInvite: DraftInvite
+  handleEditDraftInvite: (_: DraftInvite) => void
+  handleInviteUser: () => void
+  handleRemoveUser: (userId: string) => void
+  handleResendInvite: (inviteId: string) => void
+  handleWithdrawInvite: (inviteId: string) => void
+  invites: Invite[]
+  removeInviteStatus: {id: string; status: RemoteDataState}
+  removeUserStatus: {id: string; status: RemoteDataState}
+  status: RemoteDataState
+  users: CloudUser[]
+}
+
+export const draft: DraftInvite = {
+  email: '',
+  role: 'owner' as const,
+}
+
+export const DEFAULT_CONTEXT: UsersContextType = {
+  draftInvite: draft,
+  handleEditDraftInvite: (_draftInvite: DraftInvite) => {},
+  handleInviteUser: () => {},
+  handleRemoveUser: (_userId: string) => {},
+  handleResendInvite: (_inviteId: string) => {},
+  handleWithdrawInvite: (_inviteId: string) => {},
+  invites: [],
+  removeInviteStatus: {id: null, status: RemoteDataState.NotStarted},
+  removeUserStatus: {id: null, status: RemoteDataState.NotStarted},
+  status: RemoteDataState.NotStarted,
+  users: [],
+}
+
+export const UsersContext = React.createContext<UsersContextType>(
+  DEFAULT_CONTEXT
+)
+
+export const UsersProvider: FC<Props> = React.memo(({children}) => {
+  const dispatch = useDispatch()
+  const orgId = useSelector(getOrg)?.id
+  const [users, setUsers] = useState<CloudUser[]>([])
+  const [invites, setInvites] = useState<Invite[]>([])
+  const [draftInvite, setDraftInvite] = useState<DraftInvite>(draft)
+  const [status, setStatus] = useState(RemoteDataState.NotStarted)
+  const [removeInviteStatus, setRemoveInviteStatus] = useState({
+    id: null,
+    status: RemoteDataState.NotStarted,
+  })
+  const [removeUserStatus, setRemoveUserStatus] = useState({
+    id: null,
+    status: RemoteDataState.NotStarted,
+  })
+
+  const getUsersAndInvites = useCallback(async () => {
+    try {
+      const [userResp, inviteResp] = await Promise.all([
+        getOrgsUsers({orgId}),
+        getOrgsInvites({orgId}),
+      ])
+
+      if (userResp.status !== 200) {
+        throw new Error(userResp.data.message)
+      }
+
+      if (inviteResp.status !== 200) {
+        throw new Error(inviteResp.data.message)
+      }
+
+      const users =
+        userResp.data?.users?.map(u => ({
+          ...u,
+        })) ?? []
+      const invites =
+        inviteResp.data?.invites?.map(i => ({
+          ...i,
+        })) ?? []
+
+      setUsers(users)
+      setInvites(invites)
+      setStatus(RemoteDataState.Done)
+    } catch (error) {
+      setUsers([])
+      setInvites([])
+      setStatus(RemoteDataState.Error)
+      console.error(error)
+    }
+  }, [orgId])
+
+  useEffect(() => {
+    getUsersAndInvites()
+  }, [getUsersAndInvites])
+
+  const handleInviteUser = useCallback(async () => {
+    try {
+      const resp = await postOrgsInvite({orgId, data: draftInvite})
+
+      if (resp.status !== 201) {
```

Comment:
for all these `if (status !== 200)` or `if (status !== 201)` do we know that they will always be 200 or 201? I'm wondering if it's safer to do `if (status >= 300 || status < 200)`. Technically any response in the 200-299 range could be success.

Reply:
I'll check the API response, but i'd err on the side of caution and assume that any successfully response that's returned that doesn't abide by the API contract should be flagged as an error (we use this pattern throughout the app)